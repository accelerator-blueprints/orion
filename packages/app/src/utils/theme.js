export default {
  color: {
    blue: '#2165e5',
    midnightBlue: '#194cac',
    sand4: '#6d6d68',
    sand3: '#908a8a',
    sand2: '#a8a4a3',
    sand1: '#f4f4f2',
    supersplit: '#fd775e',
    brunchPink: '#fd7a9e',
    bubblegum: '#f9c3c0',
  },
  fontFamily: {
    header: 'Poppins, sans-serif',
    body: 'Didact Gothic, sans-serif',
  },
  breakpoint: {
    smUp: { min: '600px' },
    mdUp: { min: '960px' },
    lgUp: { min: '1200px' },
    xlUp: { min: '1920px' },
    smDown: { max: '600px' },
    mdDown: { max: '960px' },
    lgDown: { max: '1200px' },
    xlDown: { max: '1920px' },
  },
}
