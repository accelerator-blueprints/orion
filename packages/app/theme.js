/**
 * To switch themes, edit the lines below, and `themeName` in gatsby-config.js
 *
 * Available themes:
 *
 * 'nearform-theme'
 * 'efqm-theme'
 *
 **/
export { default } from 'efqm-theme'
export * from 'efqm-theme'
