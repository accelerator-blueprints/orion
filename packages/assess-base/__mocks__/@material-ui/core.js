import React from 'react'
const materialUi = jest.requireActual('@material-ui/core')

module.exports = {
  ...materialUi,
  // Mocking Button and IconButton prevents warnings when passing a Link component to them
  Button: jest.fn(props =>
    props.component ? (
      <a href={props.to}>{props.children}</a>
    ) : (
      <button {...props}>{props.children}</button>
    )
  ),
  IconButton: jest.fn(props =>
    props.component ? (
      <a href={props.to}>{props.children}</a>
    ) : (
      <button {...props}>{props.children}</button>
    )
  ),
}
