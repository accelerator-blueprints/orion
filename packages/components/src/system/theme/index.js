import { fontFamilies } from './typography'
import { OrangePalette, KnowledgebasePalette } from './colors'
import layout from './grid'

export const OrangeTheme = { ...fontFamilies, ...OrangePalette, ...layout }
export const KnowledgebaseTheme = {
  ...fontFamilies,
  ...KnowledgebasePalette,
  ...layout
}
