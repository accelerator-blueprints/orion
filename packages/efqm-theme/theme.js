const fontFamily = (exports.fontFamily = 'Lato, sans-serif')

const typography = {
  h1: {
    fontFamily,
    fontWeight: '900',
    fontSize: 28,
    letterSpacing: 0.5,
  },
  h2: {
    fontFamily,
    fontWeight: 'bold',
    fontSize: 18,
    lineHeight: 'normal',
    letterSpacing: -0.1,
  },
  h3: {
    fontFamily,
    fontWeight: '900',
    fontSize: 12,
    letterSpacing: 1.8,
    textTransform: 'uppercase',
  },
  h4: {
    fontFamily,
    fontWeight: 'bold',
    fontSize: 11,
    letterSpacing: 1.2,
    textTransform: 'uppercase',
  },
  h5: {
    fontFamily,
    fontWeight: 'bold',
  },
  h6: {
    // Mainly used for bold numbers in tables and graphics
    fontFamily,
    fontSize: 20,
    fontWeight: '900',
    lineHeight: 1.4,
    letterSpacing: -0.2,
  },
  body2: {
    fontFamily,
    fontSize: 14,
    letterSpacing: 'normal',
  },
  body1: {
    fontFamily,
    fontSize: 16,
    letterSpacing: 'normal',
  },
  placeholder: {
    fontFamily,
    fontSize: 14,
    lineHeight: 'normal',
    letterSpacing: -0.1,
  },
  button: {
    fontFamily,
    fontSize: 12,
    fontWeight: '900',
    lineHeight: 'normal',
    letterSpacing: 1.8,
  },
}

const efqmDigitalPalette = {
  // Names / keys are unofficial descriptions
  slateGrey: 'rgb(69, 75, 102)',
  navyBlue: 'rgb(73, 98, 173)',
  cyan: 'rgb(80, 184, 242)',
  aqua: 'rgb(121, 204, 198)',
  emeraldGreen: 'rgb(112, 214, 112)',
  pearGreen: 'rgb(155, 220, 106)',

  // Shades used in designs not included in EFQM docs
  paleGrey: 'rgb(244, 246, 248)',
  midGrey: 'rgb(152, 175, 198)',
  white: '#fff',
  black: '#000',
}

const shadows = [
  'none',
  // material ui requires  25 elevations
  ...new Array(24).fill(
    '0 0 5px 0 rgba(0, 0, 0, 0.1), 0 2px 10px 0 rgba(0, 0, 0, 0.1)'
  ),
]

exports.muiTheme = {
  overrides: {
    MuiButton: {
      root: {
        whiteSpace: 'nowrap',
        boxShadow: shadows[1],
      },
      text: {
        boxShadow: 'none',
      },
      contained: {
        padding: '9px 24px 8px',
      },
      outlinedPrimary: {
        border: 'none',
        '&:hover': {
          border: 'none',
        },
      },
      outlinedSecondary: {
        border: 'none',
        '&:hover': {
          border: 'none',
        },
      },
      outlined: {
        padding: '8px 24px 7px',
      },
      sizeSmall: {
        fontSize: 11,
      },
    },
    MuiInput: {
      root: {
        ...typography.body1,
        borderRadius: 3,
        backgroundColor: efqmDigitalPalette.paleGrey,
      },
      input: {
        padding: '7.5px 12px 8.5px',
      },
    },
    MuiOutlinedInput: {
      root: {
        ...typography.body1,
        borderRadius: 3,
        boxShadow: shadows[1],
        background: efqmDigitalPalette.white,
      },
      notchedOutline: {
        border: 'none',
      },
      input: {
        padding: '7.5px 12px 8.5px',
      },
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '.75em',
      },
    },
    MuiTableCell: {
      head: {
        ...typography.h4,
      },
      body: {
        ...typography.body1,
      },
    },
  },
  props: {
    MuiInput: {
      disableUnderline: true,
    },
    MuiPaper: {
      elevation: 1,
    },
  },
  typography: {
    fontFamily,
    fontWeight: 'normal',
    ...typography,
  },
  palette: {
    primary: {
      main: efqmDigitalPalette.navyBlue,
      light: efqmDigitalPalette.cyan,
      dark: efqmDigitalPalette.slateGrey,
    },
    secondary: {
      main: efqmDigitalPalette.aqua,
      light: efqmDigitalPalette.pearGreen,
      dark: efqmDigitalPalette.emeraldGreen,
    },
    background: {
      default: efqmDigitalPalette.white,
      paper: efqmDigitalPalette.white,
      light: efqmDigitalPalette.paleGrey,
      dark: efqmDigitalPalette.midGrey,
    },
    contrastThreshold: 1.6,
  },
  shadows,
}

exports.googleFonts = ['Lato:400,700,900']
