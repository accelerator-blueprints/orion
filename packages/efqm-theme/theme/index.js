const palette = require('./mui/palette')
const typography = require('./mui/typography')
const shadows = require('./mui/shadows')
const props = require('./mui/props')

const MuiButton = require('./mui/muiComponentsOverrides/MuiButton')
const MuiListItemText = require('./mui/muiComponentsOverrides/MuiListItemText')
const MuiOutlinedInput = require('./mui/muiComponentsOverrides/MuiOutlinedInput')
const MuiTypography = require('./mui/muiComponentsOverrides/MuiTypography')
const MuiTableCell = require('./mui/muiComponentsOverrides/MuiTableCell')
const MuiInput = require('./mui/muiComponentsOverrides/MuiInput')
const MuiInputBase = require('./mui/muiComponentsOverrides/MuiInputBase')

const articleTypography = require('./custom/articleTypography')
const articleStatusColor = require('./custom/articleStatusColor')
const articleStatusLabelColor = require('./custom/articleStatusLabelColor')
const articleStatusIconColor = require('./custom/articleStatusIconColor')
const articleRateIconColor = require('./custom/articleRateIconColor')
const articleWidgetColor = require('./custom/articleWidgetColor')
const taxonomyColor = require('./custom/taxonomyColor')
const editorsPicks = require('./custom/editorsPicks')
const iconLight = require('./custom/iconLight')
const profileSpacerBackground = require('./custom/profileSpacerBackground')

exports.muiTheme = {
  overrides: {
    MuiButton,
    MuiInput,
    MuiInputBase,
    MuiListItemText,
    MuiOutlinedInput,
    MuiTypography,
    MuiTableCell,
  },
  props,
  typography,
  palette,
  shadows,
  iconLight,
  articleTypography,
  articleStatusColor,
  articleStatusLabelColor,
  articleStatusIconColor,
  articleWidgetColor,
  articleRateIconColor,
  profileSpacerBackground,
  taxonomyColor,
  editorsPicks,
}

exports.googleFonts = ['Lato:400,700,900']
