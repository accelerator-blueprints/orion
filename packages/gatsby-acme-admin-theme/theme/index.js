const MuiListItemText = require('./mui/muiComponentsOverrides/mui-list-item-text')
const MuiIcon = require('./mui/muiComponentsOverrides/mui-icon')
const MuiDrawer = require('./mui/muiComponentsOverrides/mui-drawer')
const MuiPaper = require('./mui/muiComponentsOverrides/mui-paper')
const MuiAppBar = require('./mui/muiComponentsOverrides/mui-app-bar')
const MuiPopover = require('./mui/muiComponentsOverrides/mui-popover')
const MuiButton = require('./mui/muiComponentsOverrides/mui-button')
const MuiInput = require('./mui/muiComponentsOverrides/mui-input')

// Section below are theme key word specific overrides https://material-ui.com/customization/default-theme/
const palette = require('./mui/palette')
const typography = require('./mui/typography')

// Custom class related themes
const articleContent = require('./custom/article-content')
const articleMetadata = require('./custom/article-metadata')
const footer = require('./custom/footer')
const searchInput = require('./custom/search-input')
const secondaryAppBar = require('./custom/secondary-app-bar')

exports.muiTheme = {
  overrides: {
    MuiListItemText,
    MuiIcon,
    MuiDrawer,
    MuiPaper,
    MuiAppBar,
    MuiPopover,
    MuiButton,
    MuiInput,
  },
  palette,
  typography,
  articleContent,
  articleMetadata,
  footer: { footer },
  searchInput,
  secondaryAppBar,
}

exports.googleFonts = ['Titillium Web:400,700,900']
