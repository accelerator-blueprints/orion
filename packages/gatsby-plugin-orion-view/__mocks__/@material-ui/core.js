import React from 'react'
const materialUi = jest.requireActual('@material-ui/core')

module.exports = {
  ...materialUi,
  Button: jest.fn(props =>
    props.component ? (
      <a href={props.to}>{props.children}</a>
    ) : (
      <button type="button" {...props}>
        {props.children}
      </button>
    )
  ),
}
