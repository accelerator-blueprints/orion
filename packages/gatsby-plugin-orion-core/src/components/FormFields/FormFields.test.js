import React from 'react'
import { render } from '@testing-library/react'
import { InputField } from '.'
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles'
import theme from 'gatsby-theme-acme'

jest.mock('aws-amplify')

const renderComponent = propOverrides => {
  const props = {
    name: 'test input',
    options: ['hello', 'test', 'person'],
    error: 'mock error string',
    value: 'hello',
    ...propOverrides,
  }
  return render(
    <ThemeProvider theme={createMuiTheme(theme)}>
      <InputField {...props}>mock label name</InputField>
    </ThemeProvider>
  )
}

describe('InputField component', () => {
  describe('When the type prop is "select"', () => {
    it('composes a select component', () => {
      const { container } = renderComponent({ type: 'select' })
      expect(container).toMatchSnapshot()
    })
  })
  describe('When the type prop is "image"', () => {
    it('composes a select component', () => {
      const props = {
        path: '/a/path/to/test.jpg',
        value: 'mock value',
        aspectRatio: 1.333,
        onChange: jest.fn(),
        onUpload: jest.fn(),
        isBoxAlwaysShown: false,
        children: jest.fn(),
        isAutoFileNameEnabled: false,
      }
      const { container } = renderComponent(props)
      expect(container).toMatchSnapshot()
    })
  })
  describe('When the type prop is "tags"', () => {
    it('composes a the TagsSelect component', () => {
      const { container } = renderComponent({ type: 'tags' })
      expect(container).toMatchSnapshot()
    })
  })
})
