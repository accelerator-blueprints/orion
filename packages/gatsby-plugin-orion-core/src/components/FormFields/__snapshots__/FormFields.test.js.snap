// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component When the type prop is "image" composes a select component 1`] = `
<div>
  <h6
    class="MuiTypography-root MuiTypography-h6 MuiTypography-colorError MuiTypography-gutterBottom"
  >
    mock label name
  </h6>
  <div
    aspectratio="1.333"
    class="MuiFormControl-root MuiTextField-root makeStyles-input-87 MuiFormControl-fullWidth"
    options="hello,test,person"
    path="/a/path/to/test.jpg"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="true"
        class="MuiInputBase-input MuiInput-input"
        name="test input"
        type="text"
        value="mock value"
      />
    </div>
  </div>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorError MuiTypography-gutterBottom"
  >
    mock error string
  </p>
</div>
`;

exports[`InputField component When the type prop is "select" composes a select component 1`] = `
<div>
  <h6
    class="MuiTypography-root MuiTypography-h6 MuiTypography-colorError MuiTypography-gutterBottom"
  >
    mock label name
  </h6>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline"
    theme="[object Object]"
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby=" mui-component-select-test input"
      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-test input"
      role="button"
      tabindex="0"
    >
      Hello
    </div>
    <input
      name="test input"
      type="hidden"
      value="hello"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorError MuiTypography-gutterBottom"
  >
    mock error string
  </p>
</div>
`;

exports[`InputField component When the type prop is "tags" composes a the TagsSelect component 1`] = `
<div>
  <div
    class=" css-2b097c-container"
  >
    <div
      class=" css-yk16xz-control"
    >
      <div
        class=" css-g1d714-ValueContainer"
      >
        <div
          class=" css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          class="css-b8ldur-Input"
        >
          <div
            class=""
            style="display: inline-block;"
          >
            <input
              aria-autocomplete="list"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              id="test input"
              spellcheck="false"
              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
            <div
              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
            />
          </div>
        </div>
      </div>
      <div
        class=" css-1hb7zxy-IndicatorsContainer"
      >
        <span
          class=" css-1okebmr-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          class=" css-tlfecz-indicatorContainer"
        >
          <svg
            aria-hidden="true"
            class="css-6q0nyr-Svg"
            focusable="false"
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      <input
        name="test input"
        type="hidden"
      />
    </div>
  </div>
</div>
`;
